plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -181,8 +422,18 @@ class bp_connection_manager {
                              std::back_inserter(peers_to_drop));
          fc_dlog(self()->get_logger(), "peers to drop: ${p}", ("p", to_string(peers_to_drop)));
 
+         fc::lock_guard g(gossip_bps.mtx);
+         auto& prod_idx = gossip_bps.index.get<by_producer>();
          for (const auto& account : peers_to_drop) {
-            self()->connections.disconnect(config.bp_peer_addresses[account]);
+            if (auto i = config.bp_peer_addresses.find(account); i != config.bp_peer_addresses.end()) {
+               fc_dlog(self()->get_logger(), "disconnect to manual bp peer ${p}", ("p", i->second));
+               self()->connections.disconnect(i->second);
```

Comment:
`disconnect` takes a lock of `connections_mtx`... maybe it would also be better to call the `disconnect()`s after releasing the `gossip_bps.mtx` mutex?

Reply:
Yep.